File: fix_dates.py
import re
import sys

with open('src/data.ts', 'r') as f:
    content = f.read()

month_map = {
    'jan': '01', 'january': '01',
    'feb': '02', 'february': '02',
    'mar': '03', 'march': '03',
    'apr': '04', 'april': '04',
    'may': '05',
    'jun': '06', 'june': '06',
    'jul': '07', 'july': '07',
    'aug': '08', 'august': '08',
    'sep': '09', 'september': '09',
    'oct': '10', 'october': '10',
    'nov': '11', 'november': '11',
    'dec': '12', 'december': '12'
}

def replace_date(match):
    date_str = match.group(1)
    
    # Range of days: 18-20 Sep 2026 -> Take the start date 18 Sep 2026? Or leave it?
    # User said: "Every election must have one valid, complete election date stored as a real date value... YYYY-MM-DD"
    # Wait, the instruction says "If legacy records contain incomplete dates, preserve those records but identify them for correction. Do not guess their actual dates."
    
    # 15 Aug 2026 -> 2026-08-15
    m1 = re.match(r'^(\d{1,2})[\s,]+([a-zA-Z]+)[\s,]+(\d{4})$', date_str)
    if m1:
        d = m1.group(1).zfill(2)
        m = month_map.get(m1.group(2).lower(), '01')
        y = m1.group(3)
        return f"date: '{y}-{m}-{d}'"
        
    m2 = re.match(r'^(\d{1,2})[-–](\d{1,2})[\s,]+([a-zA-Z]+)[\s,]+(\d{4})$', date_str)
    if m2:
        d = m2.group(1).zfill(2)
        m = month_map.get(m2.group(3).lower(), '01')
        y = m2.group(4)
        return f"date: '{y}-{m}-{d}'"
        
    return match.group(0)

new_content = re.sub(r"date:\s*'([^']+)'", replace_date, content)

with open('src/data.ts', 'w') as f:
    f.write(new_content)

print("Done")

File: fix_dates.js
const fs = require('fs');
let content = fs.readFileSync('src/data.ts', 'utf8');

const monthMap = {
    'jan': '01', 'january': '01',
    'feb': '02', 'february': '02',
    'mar': '03', 'march': '03',
    'apr': '04', 'april': '04',
    'may': '05',
    'jun': '06', 'june': '06',
    'jul': '07', 'july': '07',
    'aug': '08', 'august': '08',
    'sep': '09', 'september': '09',
    'oct': '10', 'october': '10',
    'nov': '11', 'november': '11',
    'dec': '12', 'december': '12'
};

content = content.replace(/date:\s*'([^']+)'/g, (match, dateStr) => {
    let newDate = dateStr;
    // 15 Aug 2026 -> 2026-08-15
    const matchDmy = dateStr.match(/^(\d{1,2})[\s,]+([a-zA-Z]+)[\s,]+(\d{4})$/);
    if (matchDmy) {
        let d = matchDmy[1].padStart(2, '0');
        let m = monthMap[matchDmy[2].toLowerCase()] || '01';
        let y = matchDmy[3];
        return `date: '${y}-${m}-${d}'`;
    }
    // Aug 2026 -> 2026-08-01 (Wait, is this correct? The user said "do not guess their actual dates" "If legacy records contain incomplete dates, preserve those records but identify them for correction.")
    // Actually, "Do not guess their actual dates. If legacy records contain incomplete dates, preserve those records but identify them for correction."
    // Let me just replace the ones I can parse exactly and maybe keep the others for now, but wait, the instruction says:
    // "On the public website, display election dates in full as: 16 August 2026"
    // "Remove quarter-based or month/year-only public date displays where an actual complete election date exists."
    return match;
});

console.log("Done");
